src/app/selectors.js

```diff
@@ -562,12 +562,44 @@ export const getWSControllerURL = createSelector(
   Returns the controller data in the format of an Object.entries output.
   [wsControllerURL, [data]]
   @param {String} controllerUUID The full controller UUID.
+  @returns {Array} The controller data in the format of an Object.entries output.
 */
 export const getControllerDataByUUID = (controllerUUID) => {
   return createSelector(getControllerData, (controllerData) => {
     if (!controllerData) return null;
-    return Object.entries(controllerData).find(
-      (controller) => controllerUUID === controller[1][0].uuid
-    );
+    const found = Object.entries(controllerData).find((controller) => {
+      // Loop through the sub controllers for each primary controller.
+      // This is typically only seen in JAAS. Outside of JAAS there is only ever
+      // a single sub controller.
+      return controller[1].find(
```

Comment:
If the purpose of this to is to return the first controller, even if there is only one - why isn't this array zero based?

Reply:
It is the result from an `Object.entries` in which the `0` index is the key and the `1` index is the value. In this case, the list of controllers.